packages/auth/src/grant/routes.ts

```diff
@@ -26,15 +26,15 @@ interface ServiceDependencies extends BaseService {
   config: IAppConfig
 }
 
-type GrantRequest<BodyT = never, QueryT = ParsedUrlQuery> = Omit<
+type GrantRequest<BodyT = never, QueryT = ParsedUrlQuery> = Exclude<
```

Comment:
`Exclude` here because `AppContext` is a union. Using `Omit` here gives back just an empty type, minus the defined `body` and `query` fields

Reply:
:confused: afaict `AppContext` is an intersection
```
type ParameterizedContext<StateT = Koa.DefaultState, ContextT = Koa.DefaultContext, ResponseBodyT = unknown> = Koa.ExtendableContext & {
    state: StateT;
} & ContextT & {
    body: ResponseBodyT;
    response: {
        body: ResponseBodyT;
    };
}
```

Reply:
Or is it that `AppContext['request']` is a union type via the double `extends`?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/koa/index.d.ts#L651
```
    interface BaseContext extends ContextDelegatedRequest, ContextDelegatedResponse {
```

Reply:
@wilsonianb it is an intersection indeed -> but I'm having a hard time understanding why exactly Omit isn't working. I'm trying to come up with a simpler example

Reply:
Looks like its the `DefaultContext` with the `[key: string]: any` that's causing issues:


[Playground Link](https://www.typescriptlang.org/play?#code/C4TwDgpgBAQghgZwgYQPYDtgQB7CgXigG8BYAKCkqgCNEIAuKBYAJwEt0BzcqqVYABYQW8JI2bsuPKiwgBHAK4Rm41h25kAvuXKhIUAIJgwaTDjyFYdU1ly7w0IyYy3gAdTaDkC5qgC2AAosqGAIBFZINuZQAGTEUADaANYQIKqSnAC6jHDoIADcUJpQAPQl8cmp6erZULkFRVBsYYLQzQhKOmR60ADyfp5YACbh-Z4APE5RuAA0UABEsorKwPMAfLHE0pRLSipMalJaXQDGGMxNAHIQEEMAogBuwiCC6oxjwMPhpBQy8nvAegAciB5G0ZHs+g+ww8Xh8wH8QRCCAASv8VqMBsBJsZpu5PAJvL5AsFQnNFujmOtNj9eLsVtUjuDyGd0Bc2AApeHXW5o5b7aG3WGE+GI0moykWLa-HaS4Gg44QshlKA9KB3bAnAA2CiGQoJRIRJORfIB4Q12t1EBxzjMuGFhrFyPJ9KpGziPxVdLlBwy5BVzOV5VZ7IMWoQqB59yeLBeAje6s1Or1QwdouNoVNGMInvK3v5gJB-vKmiAA)

<img width="1194" alt="Screen Shot 2022-12-22 at 1 27 33 AM" src="https://user-images.githubusercontent.com/15069181/209029345-63556b7c-118a-464f-b8a1-031905db4598.png">